client/src/main/java/com/pk/lanserver/BasicUdpServer.java

```diff
@@ -0,0 +1,258 @@
+package com.pk.lanserver;
+
+import com.pk.lanserver.models.Packet;
+import com.pk.lanserver.models.Player;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Vector;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BasicUdpServer implements UdpServer {
+  private String nick;
+  private String profileImg;
+  private DatagramSocket ds;
+  private Integer remotePort;
+  private CompletableFuture<List<Player>> futureActivePlayers = null;
+  private Instant start = null;
+  private Vector<Player> activePlayers = null;
+  private Set<String> setLocalIps;
+
+  private static final Integer TIMEOUT = 4;
+
+  public BasicUdpServer(String nick, String profileImg, Integer localPort, Integer remotePort)
+      throws SocketException {
+    setNick(nick);
+    setProfileImg(profileImg);
+    this.remotePort = remotePort;
+    ds = createSocket(localPort);
+    ds.setBroadcast(true);
+    setLocalIps = new HashSet<>();
+
+    Enumeration<NetworkInterface> nets = NetworkInterface.getNetworkInterfaces();
+    for (NetworkInterface netint : Collections.list(nets)) {
+      log.info("Display name: {}", netint.getDisplayName());
+      log.info("Name: {}", netint.getName());
+      Enumeration<InetAddress> inetAddresses = netint.getInetAddresses();
+      for (InetAddress inetAddress : Collections.list(inetAddresses)) {
+        log.info("InetAddress: {}", inetAddress);
+        setLocalIps.add(inetAddress.getHostAddress());
+      }
+    }
+  }
+
+  public void setNick(String nick) {
+    if (nick.length() == 0) {
+      throw new IllegalArgumentException("Nick is empty");
+    }
+    this.nick = nick;
+  }
+
+  public void setProfileImg(String profileImg) {
+    try {
+      Base64.getDecoder().decode(profileImg);
+    } catch (IllegalArgumentException iae) {
+      throw new IllegalArgumentException("Nick is not base64 encoded");
+    }
+    if (nick.length() == 0) {
+      throw new IllegalArgumentException("Nick is empty");
+    }
+    this.profileImg = profileImg;
+  }
+
+  /** */
+  @Override
+  public Integer call() throws Exception {
+    try {
+      ds.setSoTimeout(500);
+      for (; ; ) {
```

Comment:
what is this ; ; and why not ;;

Reply:
Ask google java format, it's not my invention.